File: hello/F-MPI/hello.f90
program hello

  use mpi

  implicit none

  integer :: size, rank, ierr
  integer :: comm = MPI_COMM_WORLD

  call MPI_Init(ierr)
  call MPI_Comm_size(comm, size, ierr)
  call MPI_Comm_rank(comm, rank, ierr)

  if (rank == 0) write(*,*) "Running on ", size, " process(es)"
  write(*,*) "Hello from rank ", rank, "!"

  call MPI_Finalize(ierr)

end program hello

File: hello/C-MPI/hello.c
#include <stdio.h>
#include <mpi.h>

int main(void)
{
  int rank, size;
  MPI_Comm comm = MPI_COMM_WORLD;

  MPI_Init(NULL,NULL);
  MPI_Comm_size(comm, &size);
  MPI_Comm_rank(comm, &rank);

  if (rank == 0) printf("Running on %d process(es)\n", size);
  printf("Hello from rank %d!\n", rank);

  MPI_Finalize();

  return 0;
}
